akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -627,6 +627,75 @@ final class Broadcast[T](val outputPorts: Int, val eagerCancel: Boolean) extends
 
 }
 
+object WireTap {
+  private val singleton = new WireTap[Nothing]
+  def apply[T]() = singleton.asInstanceOf[WireTap[T]]
+}
+
+/**
+ * Fan-out the stream to two output streams - a 'main' and a 'tap' one - emitting each incoming upstream element
+ * to all downstream consumers.
+ *
+ * '''Emits when''' element is available and demand exists from the 'main' output; the element will
+ * also be sent to the 'tap' output if there is demand.
+ *
+ * '''Backpressures when''' the 'main' output backpressures
+ *
+ * '''Completes when''' upstream completes
+ *
+ * '''Cancels when''' the 'main' output cancels
+ *
+ */
+private[stream] final class WireTap[T] extends GraphStage[FanOutShape2[T, T, T]] {
```

Comment:
add `@InternalApi` here, `private[stream]` doesn't lead to any protection from Java.